SDWebImage/SDWebImageCoder.h

```diff
@@ -30,27 +30,30 @@ CG_EXTERN BOOL SDCGImageRefContainsAlpha(_Nullable CGImageRef imageRef);
 // All the method are called inside a dispatch queue and do not block main thread
 @protocol SDWebImageCoder <NSObject>
 
-@optional
 /**
- Decode the image data to image.
-
- @param data The image data to be decoded
- @param format The recognized image format
- @return The decoded image from data
+ Returns YES if this coder can decode some data. Otherwise, it should be passed to another coder
+ 
+ @param data The image data so we can look at it
+ @return YES if this coder can decode the data, NO otherwise
  */
-- (nullable UIImage *)decodedImageWithData:(nullable NSData *)data format:(SDImageFormat)format;
+- (BOOL)canDecodeData:(nullable NSData *)data;
```

Comment:
If we want to make this more generic, we do not need to ask the image format while decoding, right ? User can use our implementation or their own `sd_imageFormatForImageData` to check which image format of the data and decide the procedure.

If we do so, these protocol method should remove that `format` args:

+ `decodedImageWithData:format`
+ `decompressedImageWithImage:data:format:shouldScaleDown`(Also I think this method last one arg a little strange. Maybe a dictionary with info key is better?)
+ `incrementalDecodedImageWithData:format`

Reply:
Done - see commits dc2664d and a3f45ae

Reply:
PS: the new signature for the decompress method is `- (UIImage *)decompressedImageWithImage:(UIImage *)image data:(NSData **)data options:(NSDictionary<NSString*, NSObject*>*)optionsDict` which is indeed generic and looks like a good API design.
We need to pass `{@"SDWebImageScaleDownLargeImages": @(YES)}` for the scaleDown to be active.